src/WebJobs.Extensions.ApiHub/Common/CodeFile2.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ApiHub.Common
+{
+    // File Triggers implement this.
+    public interface IFileTriggerStrategy<TFile>
```

Comment:
these can be internal for now. We should polish them and move them to Core since they are broadly useful to any stream-based extension (including File and Blob) 

---
In reply to: [56933822](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/58#discussion_r56933822) [](ancestors = 56933822)